sample.net.openthread.shell.xg24 fails to build

### Describe the bug

Since #111619 sample.net.openthread.shell.xg24 fails to build

https://github.com/zephyrproject-rtos/zephyr/actions/runs/29389632448/job/87270159153#step:12:2461

### Regression

- [x] This is a regression.

### Steps to reproduce

`twister -p xg24_rb4187c/efr32mg24b220f1536im48 -s sample.net.openthread.shell.xg24`
or
```
cmake -GNinja -DBOARD=xg24_rb4187c/efr32mg24b220f1536im48 ../samples/net/openthread/shell  -DCONFIG_BUILD_ONLY_NO_BLOBS=y
ninja
```

### Relevant log output

```shell
/opt/toolchains/zephyr-sdk-1.0.1/gnu/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/14.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/ieee802154/libdrivers__ieee802154.a(ieee802154_silabs_efr32.c.obj): in function `sl_802154_start_tx':
/__w/zephyr/zephyr/drivers/ieee802154/ieee802154_silabs_efr32.c:2424:(.text.silabs_efr32_tx+0x296): undefined reference to `sl_rail_start_scheduled_tx'
/opt/toolchains/zephyr-sdk-1.0.1/gnu/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/14.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /__w/zephyr/zephyr/drivers/ieee802154/ieee802154_silabs_efr32.c:2435:(.text.silabs_efr32_tx+0x2c8): undefined reference to `sl_rail_start_scheduled_cca_csma_tx'
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: linux
- SDK 1.0.1
- SHA: main since 654006d521627ce218132e0d774e93de25e5bcc1

### Additional Context

_No response_

--- a/modules/hal_silabs/simplicity_sdk/src/blob_stubs.c
+++ b/modules/hal_silabs/simplicity_sdk/src/blob_stubs.c
@@ -292,6 +292,25 @@ sl_rail_status_t sl_rail_config_multi_timer(sl_rail_handle_t rail_handle, bool e
 	return SL_RAIL_STATUS_NO_ERROR;
 }
 
+sl_rail_status_t sl_rail_start_scheduled_tx(sl_rail_handle_t rail_handle,
+	uint16_t channel,
+	sl_rail_tx_options_t tx_options,
+	const sl_rail_scheduled_tx_config_t *p_scheduled_tx_config,
+	const sl_rail_scheduler_info_t *p_scheduler_info)
+{
+	return SL_RAIL_STATUS_NO_ERROR;
+}
+
+sl_rail_status_t sl_rail_start_scheduled_cca_csma_tx(sl_rail_handle_t rail_handle,
+	uint16_t channel,
+	sl_rail_tx_options_t tx_options,
+	const sl_rail_scheduled_tx_config_t *p_scheduled_tx_config,
+	const sl_rail_csma_config_t *p_csma_config,
+	const sl_rail_scheduler_info_t *p_scheduler_info)
+{
+	return SL_RAIL_STATUS_NO_ERROR;
+}
+
 #if defined(CONFIG_IEEE802154_SILABS_EFR32)
 void sl_openthread_init(void)
 {

--- a/samples/net/openthread/shell/tests.yaml
+++ b/samples/net/openthread/shell/tests.yaml
@@ -40,7 +40,6 @@ tests:
       - esp32c6_devkitc/esp32c6/hpcore
       - esp32h2_devkitm
   sample.net.openthread.shell.xg24:
-    skip: true
     build_only: true
     extra_args:
       - CONFIG_BUILD_ONLY_NO_BLOBS=y

